src/psyclone/transformations.py

```diff
@@ -563,6 +566,394 @@ def apply(self, node, options=None):
+                    # should ignore it
+                    if (forward_dep is not None and
+                            forward_dep.abs_position < taskloop.abs_position):
+                        # If we're in a blocking single region and the
+                        # dependency for any of its tasks points to its
+                        # parent single region, then its "real" next dependency
+                        # is outside of the single region. To ensure we
+                        # synchronize before this dependency, we must have a
+                        # task synchronization construct before the spawning
+                        # thread leaves the single region. It is possible that
+                        # this dependency could be handled by another
+                        # intermediary taskwait, however I can't work out a
+                        # good way to detect that now (since there is no
+                        # "end of single region" abs_position to map to).
+                        # I could store a list of all the taskloops that
+                        # require this final taskwait, and walk from each one
+                        # to see if a taskwait has been placed to satisfy their
+                        # dependency, but that does not seem elegant somehow.
+                        if forward_dep is task_region:
+                            endwaits.append(taskloop)
+                            create_endtaskwait = True
+                        forward_dep = None
+                if forward_dep is None:
+                    continue
+                # Check if the taskloop and its forward dependence are in the
+                # same serial region
+                if (taskloops[i].ancestor(OMPSerialDirective) is
+                        forward_dep.ancestor(OMPSerialDirective)):
+                    # We're in the same OMPSerialDirective so store the
+                    # position of the forward dependency
+                    dependence_position[i] = forward_dep.abs_position
+            # Forward dependency positions are now computed for this region.
+            # Next we eliminate unneccessary dependencies. We loop over
+            # each dependence. For each dependence we loop over all other
+            # taskloops whose position is before that dependence. If that
+            # dependence is fulfilled by this dependence (i.e.
+            # dependence_position[this] < dependence_position[that]) then
+            # we remove that dependence. If this dependence is fulfilled
+            # by that dependence (i.e. dependence_position[that] <
+            # dependence_position[this]) then we remove this dependence.
+            # This assumes that taskloop_positions is in ascending order,
+            # which I believe to be true by construction.
+            for i, dep_pos in enumerate(dependence_position):
+                if dep_pos is not None:
```

Comment:
Perhaps:

    if dep_pos is None:
        continue

so as to reduce the indentation of the rest of the code.